src/knx/dptconvert.cpp

```diff
@@ -1713,9 +1713,11 @@ double float16FromPayload(const uint8_t* payload, int index)
 {
     uint16_t mantissa = unsigned16FromPayload(payload, index) & 0x87FF;
     if (mantissa & 0x8000)
-        return ((~mantissa & 0x07FF) + 1.0) * -0.01 * (1 << ((payload[index] >> 3) & 0x0F));
-
-    return mantissa * 0.01 * (1 << ((payload[index] >> 3) & 0x0F));
+        //return ((~mantissa & 0x07FF) + 1.0) * -0.01 * (1 << ((payload[index] >> 3) & 0x0F));
+        return ((~mantissa & 0x07FF) + 1.0) * -0.01 * pow(2.0, ((payload[index] >> 3) & 0x0F));
```

Comment:
Is this necessary? Shifting should produce the same result. And it should be much faster

Reply:
Was just for testing, I'll delete that right away. But the way value to data point is fine or are there still problems?